### DISCUSSION TITLE
cartLinesAdd mutation does not provide enough feedback on success/failure

### DISCUSSION THREAD

**Discussion by juanpprieto** [Shopify/storefront-api-feedback#151] [Ideas & Requests]
When we call `cartLinesAdd` we get no feedback wether the line was actually added or updated.

This is problematic because it makes it very hard for merchants to update their (success/error) UI without keeping track of the previous cart and performing a comparison logic often on client-side.

It's also very hard to reliably trigger analytics events such as "add_to_cart", because it's impossible to determine if the item was successfully added/updated (without a prev cart)

## No feedback if adding beyond what is `availableForSale`
Furthermore, If a variant has 2 `availableForSale` and I add 2 everything is ok. But, if I then add another item I don't get any indication that the mutation did not update the line quantity.

## Possible solutions
1. Return a `linesAdded: Lines[] | null` along side `cart` and `userErrors`. This one would be preferred because we get back the lines that were updated. This is the payload the "add_to_cart" analytic events require
2. Return a `useError` indicating the lines was not updated/added due to availability. Less ideal as we would need to build the "linesAdded" payload on the application to pass it to analytics

**Comment by ndimatteo** [comment]
To add to this, it appears that updating subsequent line items above the available quantity results in the API dropping the quantity to `0` and leaving it in the cart.

I documented this in detail here, which was pushed as an issue with the Storefront API: https://github.com/Shopify/hydrogen/issues/824
